### PR TITLE
Fix warnings with Sphinx>4

### LINKED ISSUE #11725: DOC: Unpin max version of Sphinx when Sphinx 4.0.2 is out (and Jinja2)
**Update:** If you want to build docs in Python 3.10 or newer, see the pins in  #13587 .

Revert #11724 when sphinx-doc/sphinx#9210 is resolved. @saimn tracked it down to sphinx-doc/sphinx#8935 .

https://github.com/astropy/astropy/blob/642d8aea136d274a82928a8735cd25d6905116ed/setup.cfg#L84-L85

Also revert #12988 .

https://github.com/astropy/astropy/blob/f6569ceaeb81afef80d5a6a0a579698f1c27c6cd/setup.cfg#L103

### PR DESCRIPTION
**Pull request by saimn**
Fix #11725, also see #11763.

Close #13015

This fixes the warnings (example below) by removing the duplicate entries.

```
astropy/coordinates/builtin_frames/galactic.py:docstring of astropy.coordinates.builtin_frames.galactic.Galactic:1:
WARNING: duplicate object description of astropy.coordinates.builtin_frames.galactic.Galactic,
other instance in api/astropy.coordinates.Galactic, use :noindex: for one of them
```

### Description

This pull request is to address ...

Fixes #<Issue Number>

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Do the proposed changes actually accomplish desired goals?
- [ ] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [ ] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [ ] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [ ] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [ ] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [ ] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Comment by pllim** [comment]
Moved the milestone.

**Comment by dhomeier** [comment]
@saimn could you rebase – some of those RTD failures look like the warnings that have just been fixed for 5.0.1?

**Comment by pllim** [comment]
I rebased, though the CI might be broken due to pytest 7 and scipy 1.8.

**Comment by pllim** [comment]
These warnings in RTD log might be related:

```
/home/docs/checkouts/readthedocs.org/user_builds/astropy/checkouts/12270/docs/units/index.rst:270:<autosummary>:1: WARNING: py:obj reference target not found: astropy.units.abstractmethod
/home/docs/checkouts/readthedocs.org/user_builds/astropy/checkouts/12270/docs/units/index.rst:277:<autosummary>:1: WARNING: py:obj reference target not found: astropy.units.ABCMeta
/home/docs/checkouts/readthedocs.org/user_builds/astropy/checkouts/12270/docs/units/index.rst:447:<autosummary>:1: WARNING: py:obj reference target not found: astropy.units.function.ABCMeta
/home/docs/checkouts/readthedocs.org/user_builds/astropy/checkouts/12270/docs/units/index.rst:447:<autosummary>:1: WARNING: py:obj reference target not found: astropy.units.function.CompositeUnit
/home/docs/checkouts/readthedocs.org/user_builds/astropy/checkouts/12270/docs/units/index.rst:447:<autosummary>:1: WARNING: py:obj reference target not found: astropy.units.function.Quantity
/home/docs/checkouts/readthedocs.org/user_builds/astropy/checkouts/12270/docs/units/index.rst:447:<autosummary>:1: WARNING: py:obj reference target not found: astropy.units.function.Unit
/home/docs/checkouts/readthedocs.org/user_builds/astropy/checkouts/12270/docs/units/index.rst:447:<autosummary>:1: WARNING: py:obj reference target not found: astropy.units.function.UnitBase
/home/docs/checkouts/readthedocs.org/user_builds/astropy/checkouts/12270/docs/units/index.rst:447:<autosummary>:1: WARNING: py:obj reference target not found: astropy.units.function.UnitConversionError
/home/docs/checkouts/readthedocs.org/user_builds/astropy/checkouts/12270/docs/units/index.rst:447:<autosummary>:1: WARNING: py:obj reference target not found: astropy.units.function.UnitTypeError
/home/docs/checkouts/readthedocs.org/user_builds/astropy/checkouts/12270/docs/units/index.rst:447:<autosummary>:1: WARNING: py:obj reference target not found: astropy.units.function.UnitsError
/home/docs/checkouts/readthedocs.org/user_builds/astropy/checkouts/12270/docs/units/index.rst:467:<autosummary>:1: WARNING: py:obj reference target not found: astropy.units.function.logarithmic.CompositeUnit
/home/docs/checkouts/readthedocs.org/user_builds/astropy/checkouts/12270/docs/units/index.rst:467:<autosummary>:1: WARNING: py:obj reference target not found: astropy.units.function.logarithmic.FunctionQuantity
/home/docs/checkouts/readthedocs.org/user_builds/astropy/checkouts/12270/docs/units/index.rst:467:<autosummary>:1: WARNING: py:obj reference target not found: astropy.units.function.logarithmic.FunctionUnitBase
/home/docs/checkouts/readthedocs.org/user_builds/astropy/checkouts/12270/docs/units/index.rst:467:<autosummary>:1: WARNING: py:obj reference target not found: astropy.units.function.logarithmic.Unit
/home/docs/checkouts/readthedocs.org/user_builds/astropy/checkouts/12270/docs/units/index.rst:467:<autosummary>:1: WARNING: py:obj reference target not found: astropy.units.function.logarithmic.UnitConversionError
/home/docs/checkouts/readthedocs.org/user_builds/astropy/checkouts/12270/docs/units/index.rst:467:<autosummary>:1: WARNING: py:obj reference target not found: astropy.units.function.logarithmic.UnitTypeError
/home/docs/checkouts/readthedocs.org/user_builds/astropy/checkouts/12270/docs/units/index.rst:467:<autosummary>:1: WARNING: py:obj reference target not found: astropy.units.function.logarithmic.UnitsError
```

**Comment by pllim** [comment]
@astrojuanlu , any ideas? 😱

**Comment by pllim** [comment]
Do you think we can do something like astropy/photutils#1306 ?

File: docs/conf.py [hunk 1]
```diff
@@ -84,6 +84,8 @@
 # major.minor, call `check_sphinx_version("X.Y.Z")` here.
 check_sphinx_version("1.2.1")  # noqa: F405
 
+autosummary_generate = False
```
Comment (pllim) [inline docs/conf.py:87]: @saimn , was this experimental or part of your proposed solution for real?

**Comment by pllim** [comment]
Nope. Doesn't work now.

**Comment by saimn** [comment]
See #13048.

**Pull request closed without merging**